### PR TITLE
tests: adc_api: Use SHRT_MIN instead of 0 to mark empty buffer entries

### PR DESCRIPTION
**Pull request by anangl**
Instead of 0, which can happen to be a sampling result, use for marking
empty buffer entries a value that is not supposed to be written in the
buffer by an ADC driver. This value defaults to SHRT_MIN but can be
overriden, if needed, for particular boards.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

### CODE REVIEW

**Review by ABOSTM — APPROVED** [review]
LGTM.
Test passed on nucleo_l073rz, nucleo_l152re, nucleo_wb55rg.

**Comment by anangl** [comment]
Corrected typos in the commit message.

File: tests/drivers/adc/adc_api/src/test_adc.c [hunk 1]
```diff
@@ -269,7 +277,9 @@ static const struct device *init_adc(void)
 		"Setting up of the second channel failed with code %d", ret);
 #endif /* defined(ADC_2ND_CHANNEL_ID) */
 
-	(void)memset(m_sample_buffer, 0, sizeof(m_sample_buffer));
+	for (i = 0; i < BUFFER_SIZE; ++i) {
```
Comment (ioannisg) [inline tests/drivers/adc/adc_api/src/test_adc.c:280]: curious, why memset could not work here?

Comment (anangl) [reply tests/drivers/adc/adc_api/src/test_adc.c:280]: Because it writes bytes and the buffer items are of type `int16_t`.

Comment (ioannisg) [reply tests/drivers/adc/adc_api/src/test_adc.c:280]: wouldn't sizeof operator take care of this?

Comment (anangl) [reply tests/drivers/adc/adc_api/src/test_adc.c:280]: It's not about the size but the value written. `memset()` was okay for writing all zeros, but you cannot write non-zero 16-bit values with it.

Comment (ioannisg) [reply tests/drivers/adc/adc_api/src/test_adc.c:280]: Right, now I get it. You are actually checking the SHORT MIN values at run time; i thought int was only needed to set the m_sample_buffer contents  initially to non-zero

**Review by ioannisg — APPROVED** [review]

**Pull request merged**